__tests__/unit/core-snapshots/filesystem/stream-reader-writer.test.ts

```diff
@@ -0,0 +1,188 @@
+import "jest-extended";
+
+import pluralize from "pluralize";
+import { pascalize } from "xcase";
+
+import { Readable } from "stream";
+import { dirSync, setGracefulCleanup } from "tmp";
+import {decamelize} from "xcase";
+
+import { Container } from "@packages/core-kernel";
+import { Sandbox } from "@packages/core-test-framework/src";
+
+import { StreamReader, StreamWriter } from "@packages/core-snapshots/src/filesystem";
+import { Identifiers } from "@packages/core-snapshots/src/ioc";
+import * as Contracts from "@packages/core-snapshots/src/contracts";
+import { JSONCodec, MessagePackCodec } from "@packages/core-snapshots/src/codecs";
+import * as Exceptions from "@packages/core-snapshots/src/exceptions";
+
+import { Assets } from "../__fixtures__";
+
+
+class DbStream extends Readable {
+    private count = 0;
+    private prefix: string = "";
+
+    constructor(private table: string) {
+        super({objectMode: true});
+
+        switch (table) {
+            case "blocks":
+                this.prefix = "Block_"
+                break;
+            case "transactions":
+                this.prefix = "Transaction_"
+                break
+            case "rounds":
+                this.prefix = "Round_"
+                break;
+        }
+    }
+
+    _read() {
+        if (this.count !== Assets[this.table].length) {
+            this.push(this.appendPrefix(Assets[this.table][this.count]));
+            this.count++;
+        }
+        else {
+            this.push(null)
+        }
+    }
+
+    private appendPrefix(entity: any) {
+        let itemToReturn = {};
+
+        let item = entity;
+
+        for(let key of Object.keys(item)) {
+            itemToReturn[this.prefix + decamelize(key)] = item[key];
+        }
+
+        return itemToReturn;
+    }
+}
+
+const getSingularCapitalizedTableName = (table: string) => {
+    return pascalize(pluralize.singular(table));
+}
+
+const getEncode = (table, codec) => {
+    if (codec === "json") {
+        return new JSONCodec()[`encode${getSingularCapitalizedTableName(table)}`];
+    } else {
```

Comment:
No need for the `else` as the next line will never be reached if the previous `if` is true.